### ISSUE TITLE
Dynamic Fields

### ISSUE THREAD

**Issue by aydreeihn** [osTicket/roadmap#5]
**Feature Summary**

We need to build functionality that will be able to render a form with fields based on the information saved in the database.

**Intended Outcome of the feature**

A form and all of its fields will be retrieved in JSON format from the database. We will pull form and field configurations to determine the type of field as well as how it should be styled on the form and then render it as desired in osTicket. This will also act as a first step towards being able to create a drag and drop form builder in the future.

**How will it work once completed?**

When a page that includes a form is navigated to, we will retrieve the form and its fields from the backend and render them on the page based on the configurations saved in the database.

**Comment by GrahamSH-LLK** [comment]
Is this for OSTicket v2? Speaking of that, how's it going?

**Comment by aydreeihn** [comment]
@GrahamSH-LLK Yes, this is for 2.0. It's going well so far! We have a nice infrastructure set up so far for Laravel and ReactJS. Now that we have a good structure in place for 2.0, we will be tracking our progress more through Github using Issues and Projects. This will give the community more visibility on where we're at as well!

**Comment by GrahamSH-LLK** [comment]
> @GrahamSH-LLK Yes, this is for 2.0. It's going well so far! We have a nice infrastructure set up so far for Laravel and ReactJS. Now that we have a good structure in place for 2.0, we will be tracking our progress more through Github using Issues and Projects. This will give the community more visibility on where we're at as well!

Awesome. I assume you aren't open-sourcing it until much later in development? Can't wait for 2.0!

**Comment by protich** [comment]
Field dependency and visibility 

- Field visibility based on another field 
- Field selection or option based on another field selection etc. 

Support dynamic selection / options 

-  Support typeahead selection
